### PR TITLE
Add i8*i4 and i8*i8 matmul microbenchmark

### PR DESCRIPTION
**Pull request by mariecwhite**

### CODE REVIEW

File: build_tools/python/e2e_test_framework/models/model_groups.py [hunk 1]
```diff
@@ -26,6 +26,18 @@
...
+        model=matmul.MATMUL_1x256x2048_I8xI4_MLIR, threads=[0, 1]
+    ),
+    common_definitions.CpuBenchmarkConfig(
+        model=matmul.MATMUL_256x256x2048_I8xI4_MLIR, threads=[0, 1]
+    ),
+    common_definitions.CpuBenchmarkConfig(
+        model=matmul.MATMUL_1x256x2048_I8xI8_MLIR, threads=[0, 1]
+    ),
+    common_definitions.CpuBenchmarkConfig(
+        model=matmul.MATMUL_256x256x2048_I8xI8_MLIR, threads=[0, 1]
```
Comment (pzread) [inline build_tools/python/e2e_test_framework/models/model_groups.py]: Do we need to test local-sync and 1-thread. I feel like for microbenchmark local-sync might be enough?

File: build_tools/python/e2e_test_framework/models/matmul.py [hunk 2]
```diff
@@ -107,3 +107,43 @@
     entry_function="matmul_123x2561x2561_f32t_f32t_f32t_tile_config_default",
     input_types=["123x2561xf32", "2561x2561xf32", "123x2561xf32"],
 )
+
+MATMUL_1x256x2048_I8xI4_MLIR = common_definitions.Model(
+    id=unique_ids.MICRO_MATMUL_1x256x2048_I8xI4_MLIR,
+    name="matmul_1x256x2048_i8_i4_i32_tile_config_default",
+    tags=["i8i4", "ubench", "matmul"],
+    source_type=common_definitions.ModelSourceType.EXPORTED_STABLEHLO_MLIR,
+    source_url="https://storage.googleapis.com/iree-model-artifacts/jax/jax_models_0.4.25_1710887127/DOT_PRODUCT_JAX_1X256X2048XI8I4/stablehlo.mlirbc",
```
Comment (pzread) [inline build_tools/python/e2e_test_framework/models/matmul.py]: I'm a little curious why we chose stablehlo? To me it seems to be easier to represent microbenchmarks in linalg? (and we can skip the import stage during compilation)

File: build_tools/python/benchmark_suites/iree/armv8_a_benchmarks.py [hunk 3]
```diff
@@ -24,6 +24,10 @@ class Android_ARMv8_A_Benchmarks(object):
         tf_models.GPT2_117M_1x1_FP32_TF,
         tflite_models.MOBILEBERT_INT8,
         tflite_models.VIT_INT8_TFL,
+        matmul.MATMUL_1x256x2048_I8xI4_MLIR,
+        matmul.MATMUL_256x256x2048_I8xI4_MLIR,
+        matmul.MATMUL_1x256x2048_I8xI8_MLIR,
+        matmul.MATMUL_256x256x2048_I8xI8_MLIR,
```
Comment (pzread) [inline build_tools/python/benchmark_suites/iree/armv8_a_benchmarks.py:33]: These models will be tested with local-sync, 1 and 2 threads. Maybe we want to add them separately and only test with local-sync?

File: build_tools/python/e2e_test_framework/models/model_groups.py (hunk 1 above)
Comment (hanhanW) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: +1

Comment (mariecwhite) [reply build_tools/python/e2e_test_framework/models/model_groups.py]: Good point, I updated to use local-sync only.

File: build_tools/python/e2e_test_framework/models/matmul.py (hunk 2 above)
Comment (mariecwhite) [reply build_tools/python/e2e_test_framework/models/matmul.py]: I generated these from JAX and lowered to MLIR with StableHLO. Updated to use Linalg instead.

File: build_tools/python/benchmark_suites/iree/armv8_a_benchmarks.py (hunk 3 above)
Comment (mariecwhite) [reply build_tools/python/benchmark_suites/iree/armv8_a_benchmarks.py:33]: Changed to use local-sync

**Review by pzread — APPROVED** [review]

**Comment by bjacob** [comment]
@mariecwhite , this was merged as a rebase-and-merge instead of the usual squash-and-merge. GitHub tends to remember your last choice as the default for future PRs so make sure it's squash-and-merge.

**Pull request merged**